src/form/field/GeocoderComboBox.js

```diff
@@ -0,0 +1,327 @@
+
+    /**
+     * The field to display in the combobox result. Default is
+     * "name" for instant use with the default store for this component.
+     *
+     * @cfg {String}
+     */
+    displayField: 'name',
+
+    /**
+     * The field in the GeoCoder service repsonse to be used as mapping for the
+     * 'name' field in the #store.
+     * Ignored when a store is passed in.
+     *
+     * @cfg {String}
+     */
+    displayValueMapping: 'display_name',
+
+    /**
+     * Field from selected record to use when the combo's
+     * #getValue method is called. Default is "extent". This field is
+     * supposed to contain an ol.Extent.
+     * By setting this to 'coordinate' a field holding an ol.Coordinate is used.
+     *
+     * @cfg {String}
+     */
+    valueField: 'extent',
+
+    /**
+     * The query parameter for the user entered search text.
+     * Default is 'q' for instant use with OSM Nominatim.
+     *
+     * @cfg {String}
+     */
+    queryParam: 'q',
+
+    /**'Search'
+     * Text to display for an empty field.
+     *
+     * @cfg {String}
+     */
+    emptyText: 'Search a location',
+
+    /**
+     * Minimum number of entered characters to trigger a search.
+     *
+     * @cfg {Number}
+     */
+    minChars: 3,
+
+    /**
+     * Delay before the search occurs in ms.
+     *
+     * @cfg {Number}
+     */
+    queryDelay: 100,
+
+    /**
+     * URL template for querying the geocoding service. If a store is
+     * configured, this will be ignored. Note that the #queryParam will be used
+     * to append the user's combo box input to the url.
+     *
+     * @cfg {String}
+     */
+    url: 'https://nominatim.openstreetmap.org/search?format=json',
+
+    /**
+     * The SRS used by the geocoder service.
+     *
+     * @cfg {String}
+     */
+    srs: 'EPSG:4326',
+
+    /**
+     * Zoom level when zooming to a location (#valueField='coordinate')
+     * Not used when zooming to extent.
+     *
+     * @cfg {Number}
+     */
+    zoom: 10,
+
+    /**
+     * Flag to steer if selected address feature is drawn on #map
+     * (by #locationLayer).
+     *
+     * @cfg {Boolean}
+     */
+    showLocationOnMap: true,
+
+    /**
+     * @private
+     */
+    initComponent: function() {
+        var me = this;
+
+        if (!me.store) {
+            me.store = Ext.create('Ext.data.JsonStore', {
+                fields: [
+                    {name: 'name', mapping: me.displayValueMapping},
+                    {name: 'extent', convert: me.convertToExtent},
+                    {name: 'coordinate', convert: me.convertToCoordinate}
+                ],
+                proxy: {
+                    type: 'ajax',
+                    url: me.url,
+                    reader: {
+                        type: 'json'
+                    }
+                }
+            });
+        }
+
+        if (!me.locationLayer) {
+            me.locationLayer = new ol.layer.Vector({
+                source: new ol.source.Vector()
```

Comment:
~~Maybe we can add a user-configurable style to the layer?~~
But its not really necessary as the user can configure the whole layer by himself, so just ignore the comment

Reply:
Thanks for the hint. I added a config property so you can influence the style even if the layer is auto-created. Might be handy at some point.